device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionStateCallback.java

```diff
@@ -52,7 +52,25 @@
      * closed unexpectedly.
      *
      * @param errorCondition the condition of the session that caused the close if the session closed remotely, or the condition
-     *                       of the link that closed unexpectedly
+     *                       of the link that closed unexpectedly.
+     * @param deviceId the device that the session belonged to.
      */
-    void onSessionClosedUnexpectedly(ErrorCondition errorCondition);
+    void onSessionClosedUnexpectedly(ErrorCondition errorCondition, String deviceId);
+
+    /**
+     * Executed if a session closes, but it was expected. Likely due to user calling close on the connection, or
+     * unregistering a device from an active multiplexed connection
+     *
+     * @param deviceId the device whose connection closed.
+     */
+    void onSessionClosedAsExpected(String deviceId);
```

Comment:
same for these guys, if this is only applicable to multiplexed sessions, it would be nice to have that in the name.

Reply:
This one isn't limited to multiplexed connections. Even in single device connections, this callback will be fired during cleanup